syntax = "proto3";
package main;

option go_package = "./simio";

message AnglersimInput {
    double runTime = 1;
    double timeScale = 2;
    int64 numReals = 3;
    int64 numSpecies = 4;
    int64 seed = 5;

    message FishPopType {
        double initCount = 1;
        double birthRate = 2;
        double densDepPower = 3;
        double deathRate = 4;
        double predationRate = 5;
        double predatorBirthIncRate = 6;
        double fishingRate = 7;
        repeated double predatorMatrixRow = 8; 
        repeated double preyMatrixRow = 9;
    }

    repeated FishPopType fishPops = 6;
}

message AnglersimOutput {
    repeated int64 species = 1;
    repeated double counts = 2;
}


